File: app/src/main/java/pl/pregiel/workwork/exceptions/TooShortFieldException.java
package pl.pregiel.workwork.exceptions;


import android.content.Context;
import android.widget.Toast;

import pl.pregiel.workwork.R;

public class TooShortFieldException extends ShowToastException  {
    private int minimumCharAmount = 0;

    public TooShortFieldException(String message, int minimumCharAmount) {
        super(message);

        this.minimumCharAmount = minimumCharAmount;
    }

    @Override
    public void showToast(Context context) {
        Toast.makeText(context, context.getString(R.string.error_tooShortField, getMessage(), minimumCharAmount),
                Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/pl/pregiel/workwork/exceptions/TooShortFieldException.kt
package pl.pregiel.workwork.exceptions


import android.content.Context
import android.widget.Toast

import pl.pregiel.workwork.R

class TooShortFieldException(message: String, private val minimumCharAmount: Int) : ShowToastException(message) {

    override fun showToast(context: Context) {
        Toast.makeText(context, context.getString(R.string.error_tooShortField, message, minimumCharAmount),
                Toast.LENGTH_SHORT).show()
    }
}
